cli/src/highlight.rs

```diff
@@ -0,0 +1,309 @@
+use crate::error::Result;
+use crate::loader::Loader;
+use ansi_term::{Color, Style};
+use lazy_static::lazy_static;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::{fmt, fs, io, path};
+use tree_sitter::{Language, PropertySheet};
+use tree_sitter_highlight::{highlight, highlight_html, HighlightEvent, Properties, Scope};
+
+lazy_static! {
+    static ref CSS_STYLES_BY_COLOR_ID: Vec<String> =
+        serde_json::from_str(include_str!("../vendor/xterm-colors.json")).unwrap();
+}
+
+pub struct Theme {
+    ansi_styles: Vec<Option<Style>>,
+    css_styles: Vec<Option<String>>,
+}
+
+impl Theme {
+    pub fn load(path: &path::Path) -> io::Result<Self> {
+        let json = fs::read_to_string(path)?;
+        Ok(Self::new(&json))
+    }
+
+    pub fn new(json: &str) -> Self {
+        let mut ansi_styles = vec![None; 30];
+        let mut css_styles = vec![None; 30];
+        if let Ok(colors) = serde_json::from_str::<HashMap<Scope, Value>>(json) {
+            for (scope, style_value) in colors {
+                let mut style = Style::default();
+                parse_style(&mut style, style_value);
+                ansi_styles[scope as usize] = Some(style);
+                css_styles[scope as usize] = Some(style_to_css(style));
+            }
+        }
+        Self {
+            ansi_styles,
+            css_styles,
+        }
+    }
+
+    fn ansi_style(&self, scope: Scope) -> Option<&Style> {
+        self.ansi_styles[scope as usize].as_ref()
+    }
+
+    fn css_style(&self, scope: Scope) -> Option<&str> {
+        self.css_styles[scope as usize].as_ref().map(|s| s.as_str())
+    }
+}
+
+impl Default for Theme {
+    fn default() -> Self {
+        Theme::new(
+            r#"
+            {
+              "attribute": {"color": 124, "italic": true},
+              "comment": {"color": 245, "italic": true},
+              "constant.builtin": {"color": 94, "bold": true},
+              "constant": 94,
+              "constructor": 136,
+              "embedded": null,
+              "function.builtin": {"color": 26, "bold": true},
+              "function": 26,
+              "keyword": 56,
+              "number": {"color": 94, "bold": true},
+              "property": 124,
+              "operator": {"color": 239, "bold": true},
+              "punctuation.bracket": 239,
+              "punctuation.delimiter": 239,
+              "string.special": 30,
+              "string": 28,
+              "tag": {"color": 18},
+              "variable.builtin": {"bold": true}
+            }
```

Comment:
The CLI will load an optional *theme* file at `~/.tree-sitter/theme.json` that looks like this. The keys are the scopes, and the values are ansi color numbers.